sqlglot/optimizer/rules/qualify_columns.py

```diff
@@ -0,0 +1,391 @@
+import dataclasses
```

Comment:
sorry to nit pick here a bit, but do we need the rules folder? i'd say we can keep rules if we plan to have other stuff in optimizer -- not sure if we will? maybe logical planning? if not, i'd just leave it on the optimizer level

Reply:
I did this because the optimizer functions have the same name as the file.

`optimizer/qualify_columns.py`
```python
def qualify_columns(): ...
```

`optimizer/__init__.py`
```python
from optimizer import qualify_columns

def optimize():
    qualify_columns()
```

`some_other_module.py`
```python
from optimizer import qualify_columns  # <- Is this the module or the function?
```

Other options:
1. change the function names
2. change the file names
3. don't put code in `__init__.py`
4. add aliases on the imports in `__init__.py`

Reply:
what's your preference? i like doing from optimizer import optimize

Reply:
A `rules` submodule is my preference 😛 

I understand your approach though - ruthless simplicity.

I'll move `optimize` to its own file, but import it in `__init__`.
That should keep things flat and get "from optimizer import optimize"